src/WordSearch/WordSearchSolver.ts

```diff
@@ -0,0 +1,96 @@
+import trie = require('trie-prefix-tree');
+import {WordSearchPoint} from './WordSearchPoint';
+import {WordSearchResult} from './WordSearchResult';
+
+export class WordSearchSolver {
+  private _matrix: string[][];
+  private _directions: number[][];
+
+  private _trie: any;  // tslint:disable-line
+
+  constructor(matrix: string[][]) {
+    this._matrix = matrix;
+
+    this._directions = [
+      // Cardinal directions
+      [0, 1], [-1, 0], [1, 0], [0, -1],
+      // Diagonal
+      [1, 1], [-1, -1], [1, -1], [-1, 1]
+    ];
+
+    this._trie = trie([]);
+  }
+
+  findWords(words: string[]): WordSearchResult[] {
+    for (const word of words) {
+      if (word !== '') {
+        this._trie.addWord(word);

```

Comment:
Consider trimming whitespace.  You could also think about validating that all characters are in an expected set (alphanumeric?), but in the name of i18n that may have edge cases I haven't thought about.